packages/ui/animations/Collapse.tsx

```diff
@@ -0,0 +1,61 @@
+import { Box, BoxProps } from '@chakra-ui/react';
+import { slideFadeIn, slideFadeOut, slideIn, slideOut } from './keyframes';
+import { useSize } from './useSize';
+import { useCSSTransition } from './useCSSTransition';
+
+type CollapseProps = BoxProps & {
+   show: boolean | undefined | null;
+   animateOpacity?: boolean;
+};
+
+export function Collapse({
+   show = false,
+   animateOpacity = false,
+   sx,
+   children,
+   ...otherProps
+}: CollapseProps) {
+   const transition = useCSSTransition({ show });
+   const [ref, size] = useSize();
+   const height = size?.height;
+
+   return (
+      <Box
+         className={
+            show
+               ? height == null
+                  ? undefined
+                  : transition.state
+               : transition.state
+         }
+         hidden={transition.state === 'hide'}
+         sx={{
+            ...sx,
+            '--slide-height': `${height}px`,
+            '&.hide': {
```

Comment:
Is `props.sx` intended to be an override, or a mechanism to add additional styling? Destructuring it first prevents the caller from overriding the default styling, but perhaps that is intentional.

(Relevant to `Fade.tsx` too)

Reply:
Yes intentional, albeit not even necessary for now. I'd say let's not worry about this now and come back to this if we ever need to